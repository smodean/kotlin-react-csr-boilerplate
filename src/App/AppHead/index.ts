import { memo } from 'react';

import AppHead from './AppHead.container';

export default memo(AppHead);
