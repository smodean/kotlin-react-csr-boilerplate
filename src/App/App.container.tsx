import React, { FC } from 'react';
import { Provider } from 'react-redux';

import Routes from '@Routes';

import configureStore from '@store/configureStore';

import AppHead from './AppHead';

const { store } = configureStore();

const App: FC = () => (
  <Provider store={store}>
    <AppHead />
    <Routes />
  </Provider>
);

export default App;
