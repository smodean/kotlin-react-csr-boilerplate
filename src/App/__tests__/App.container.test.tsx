import React from 'react';
import TestRenderer from 'react-test-renderer';

import App from '@App';

describe('src/App/App.container.tsx', () => {
  test('snapshot', () => {
    const appRender = TestRenderer.create(<App />);

    const tree = appRender.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('children', () => {
    const appRender = TestRenderer.create(<App />);
    const appInstance = appRender.root;

    expect(appInstance.children.length).toBe(1);
  });
});
