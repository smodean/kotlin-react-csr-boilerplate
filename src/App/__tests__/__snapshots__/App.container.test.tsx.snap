// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/App/App.container.tsx snapshot 1`] = `null`;
